service/web/publisherserver.go

```diff
@@ -29,13 +35,26 @@ func newPublisherServer(projectCore project.ICore, brokerStore brokerstore.IBrok
 
 // Produce messages to a topic
 func (s publisherServer) Publish(ctx context.Context, req *metrov1.PublishRequest) (*metrov1.PublishResponse, error) {
-	logger.Ctx(ctx).Infow("produce request received", "req", req.Topic)
+	logger.Ctx(ctx).Infow("PublishServer: produce request received", "req", req.Topic)
 	span, ctx := opentracing.StartSpanFromContext(ctx, "PublisherServer.Publish", opentracing.Tags{
 		"topic": req.Topic,
 	})
 	defer span.Finish()
-
-	if ok, err := s.topicCore.ExistsWithName(ctx, req.Topic); err != nil {
+	var appConfig *registry.Config
+	configreader.NewDefaultConfig().Load(app.GetEnv(), &appConfig)
+
+	r, _ := registry.NewRegistry(appConfig)
+	nodeBindingCore := nodebinding.NewCore(nodebinding.NewRepo(r))
+	topicCore := topic.NewCore(topic.NewRepo(r), s.projectCore, s.brokerStore)
+	publisherTask, _ := tasks.NewPublisherTask(
+		uuid.New().String(),
+		r,
+		topicCore,
+		nodeBindingCore,
+	)
+	if publisherTask.CheckIfTopicExists(ctx, req.Topic) {
+		logger.Ctx(ctx).Infow("PublishServer: Topic exists inside the cache..", "req", req.Topic)
```

Comment:
this will increase logs by no of publish requests. Do we really need this data in sumologic..?